### ISSUE TITLE
training_epoch needs to return a "loss" key in the dict

### ISSUE THREAD

**Issue by rpatrik96** [PyTorchLightning/pytorch-lightning#1175]
## 📚 Documentation

Hi everyone!

In the docs detailing the usage of the logging functions `training_epoch_end`, if `"loss": loss` is not explicitly passed as a return value, then the code will fail.

The docs at https://pytorch-lightning.readthedocs.io/en/latest/experiment_reporting.html#log-metrics is not correct, 
```python
def training_epoch_end(self, outputs):
   loss = some_loss()
   ...

   logs = {'train_loss': loss}
   results = {'log': logs}
   return results
```
shall be changed to
```python
def training_epoch_end(self, outputs):
   loss = some_loss()
   ...

   logs = {'train_loss': loss}
   results = {'loss': loss, 'log': logs} # <------------------------ Here is the change
   return results
```

*Use case:* I want to monitor the training metrics to check whether my network is able to overfit the data (for this functionality, refer to #1076 )

**Comment by Borda** [comment]
thx, @rpatrik96 mind send a PR fix?

**Comment by rpatrik96** [comment]
@Borda: Not at all, but before that, another question: the docs refers to `*_epoch_end`, but it is just `*_end`, where * is an element of `{training, validation, test}`. Will this be the name in the new release or should I fix those as well?

**Comment by Borda** [comment]
we have deprecated the simple `*_end` as you can see in changelog
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md#changed-1

**Comment by awaelchli** [comment]
`training_epoch_end` does not exist yet, right (#1076)? When you say the code fails, you probably used the old `training_end` which was on batches. 

I don't see why the loss key should be mandatory in `training_epoch_end`. As far as I know, `validation_epoch_end` also does not require a loss key in return dict.

**Comment by rpatrik96** [comment]
Oh, you are right, I was using the old version; interestingly, I got no warning even on a server using the newest lightning version. Is it any way to enable explicit warnings about to be deprecated features. I like to maintain my code to be possibly up to date - although, it does not happen all the time, as in this case. Anyway, I am closing this issue.
